typing/env.ml

```diff
@@ -2846,6 +2853,36 @@ let add_exclave_lock env = add_lock Exclave_lock env
 
 let add_unboxed_lock env = add_lock Unboxed_lock env
 
+let add_local_env_lock env = add_lock Local_env_lock env
+
+let add_quotation_lock env =
+  add_lock Quotation_lock {env with stage = env.stage + 1}
+
+let add_splice_lock env = add_lock Splice_lock {env with stage = env.stage - 1}
+
+let without_open_quotations env = env.stage = 0
+let has_open_quotations env = env.stage <> 0
```

Comment:
Having seen all the call sites of this function, I think it's better to expose a function `Env.check_no_open_quotations` that will raise error if `stage > 0`.